### ISSUE TITLE
Json::reformat edge case

### ISSUE THREAD

**Issue by TamasSzigeti** [coduo/php-matcher#200]
Encountered an interesting edge case with the latest version.

We are generating a hash with `uniqid()` and testing it as part of a json with
```
…
                "id": "@string@.matchRegex('/[a-z0-9]{13}/')",
…
```

Now in certain string literals like `5e61188283825` when it goes through Json::reformat() it fails, as the json_decode returns float infinity – interprets it as an exponential number. And then json_encode returns false which doesn't satisfy the return type, so error is thrown.

**Comment by norberttech** [comment]
Could you provide an example in sandbox https://php-matcher.norbert.tech/ so I can look into it?

**Comment by TamasSzigeti** [comment]
https://php-matcher.norbert.tech/?form%5Bvalue%5D=%7B%22something%22%3A+%225e61188283825%22%7D&form%5Bpattern%5D=%7B%22something%22%3A+%22%40string%40%22%7D&form%5Bmatch%5D=

**Comment by norberttech** [comment]
Sorry for the long waiting time, it took me a while to find some time for php-matcher but it should be fine now. 
This fix will be released in 5.0.0 version

**Comment by TamasSzigeti** [comment]
Thank you!

**Comment by norberttech** [comment]
I just released version 4.0.1 with this fix and update https://php-matcher.norbert.tech website to use the lastest version of php-matcher
